File: Models/Harvest.swift
import SwiftData

@Model
final class Harvest {
    var id: UUID
    var cropName: String
    var quantity: Double
    var unit: String
    var date: Date
    var notes: String?

    @Relationship(deleteRule: .cascade, inverse: \Plant.harvests)
    var plant: Plant?

    init(
        cropName: String,
        quantity: Double,
        unit: String,
        date: Date = Date(),
        notes: String? = nil,
        plant: Plant? = nil
    ) {
        self.id = UUID()
        self.cropName = cropName
        self.quantity = quantity
        self.unit = unit
        self.date = date
        self.notes = notes
        self.plant = plant
    }
}

File: app/src/main/java/com/soilandsupper/domain/model/Harvest.kt
package com.soilandsupper.domain.model

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "harvests",
    indices = [Index(value = ["plantId"])],
    foreignKeys = [
        ForeignKey(
            entity = Plant::class,
            parentColumns = ["id"],
            childColumns = ["plantId"],
            onDelete = ForeignKey.SET_NULL
        )
    ]
)
data class Harvest(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val plantId: Long? = null,
    val cropName: String,
    val quantity: Double,
    val unit: String,
    val date: Long = System.currentTimeMillis(),
    val notes: String = ""
)
